src/box/lua/config/configdata.lua

```diff
@@ -316,8 +319,16 @@ local mt = {
 local function validate_names(saved_names, config_names)
     -- Snapshot always has replicaset uuid and
     -- at least one peer in _cluster space.
-    assert(saved_names.replicaset_uuid)
-    assert(saved_names.instance_uuid)
+    if saved_names.replicaset_uuid == nil then
+        error(('Snapshot file %s has no "replicaset_uuid" key in _cluster ' ..
+            'system space. The snapshot is likely corrupted.'):format(
+            saved_names.snap_path), 0)
```

Comment:
Maybe `snapshot.get_path()`. I see no reason to save `snap_path` inside names, it's not a name and it's not needed anywhere, but this message, which is hard to reproduce.

Reply:
Ohh, I completely forgot, that it requires `iconfig` for that. The alternative is to add this checks to `snapshot.get_names()` or `configdata.find_saved_names`. It's actually an error while getting snapshot names, not really a validation error.

It's completely up to you, I'm ok with the diff too

Reply:
I've considered this variant, but than decided that there is some sense in keeping all the uuids/names related validation in one place. TBH, I don't see much difference.